### PR TITLE
Add UCX

### PR DESCRIPTION
**Pull request by jakirkham**
This builds a package for [UCX]( https://github.com/openucx/ucx ), which provides an optimized communication layer that can be leveraged by other libraries like OpenMPI and [Dask]( https://blog.dask.org/2019/06/09/ucx-dgx ). As such this can be incredibly useful for enabling high speed networking in HPC environments.

This relies on a few CDTs that have already been integrated to provide `libibverbs`, `libnuma`, and `librdma`.

Currently this is only building the CPU part of the package. Would like to sort out the GPU bits in the feedstock. This will rely on solving issue ( https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/237 ). That said, UCX provides a good use case for that work.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ucx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      home: https://github.com/openucx/ucx
+      license: BSD-3-Clause
+      license_family: BSD
+      license_file: LICENSE
+      summary: Unified Communication X.
+
+extra:
+  recipe-maintainers:
+    - jakirkham
+    - TomAugspurger
```
Comment (jakirkham) [inline recipes/ucx/meta.yaml]: @TomAugspurger, are you ok being listed as a maintainer here?

**Comment by TomAugspurger** [comment]
Sure.

**Comment by jakirkham** [comment]
I don't think most of the linter's complaints are relevant for this recipe. Have itemized and responded to them below.

> For **recipes/numactl-cos6-x86_64**:
> 
> * license_file entry is missing, but is expected.
> 
> For **recipes/numactl-devel-cos6-x86_64**:
> 
> * license_file entry is missing, but is expected.

These are issues being reported for PR ( https://github.com/conda-forge/staged-recipes/pull/8632 ). Will ignore them here.

> For **recipes/ucx**:
> 
> * The home item is expected in the about section.
> * The license item is expected in the about section.
> * The summary item is expected in the about section.
> * The recipe must have a `build/number` section.

This appears to be noting the lack of a top-level `about` section. However each package has its own `about` section. So this can be ignored. Have raised issue ( https://github.com/conda-forge/conda-smithy/issues/1124 ) about it.

> For **recipes/ucx**:
> 
> * It looks like the 'ucx-proc' output doesn't have any tests.

This is a metapackage. I'm not sure there is much value in adding a test here. Though suppose we could add a dummy one to make the linter happy.

**Comment by jakirkham** [comment]
Also note that this is Linux only. So no other platforms are relevant.

**Comment by jakirkham** [comment]
@conda-forge/staged-recipes, this is ready for review 😄

File: recipes/ucx/meta.yaml (hunk 1 above)
Comment (mrocklin) [reply recipes/ucx/meta.yaml]: Should we include @akshay-venkatesh or one of the UCX maintainers here as well?  This is for the C/C++ library, correct?

Comment (Akshay-Venkatesh) [reply recipes/ucx/meta.yaml]: I can definitely pitch in when it comes to the process of actually building the UCX C library.

Comment (jakirkham) [reply recipes/ucx/meta.yaml]: Yeah that would be great!

Yes this is for the C/C++ library.

The Python bits get added in as a split package as part of this recipe through some other changes I have. Will hopefully be clearer once we get this through and can add PRs to feedstock.

Comment (jakirkham) [reply recipes/ucx/meta.yaml]: Have added you @Akshay-Venkatesh 🙂 If there's anyone else from UCX you think should be  added here, please let me know and will add them too.

**Comment by jakirkham** [comment]
This now passes! 🎉

@conda-forge/staged-recipes, please take a look. 😄

**Pull request merged**